deploy.py

```diff
@@ -20,22 +20,129 @@
 from __future__ import print_function
 
 import argparse
+import collections
 import copy
 import os
 import shutil
 import subprocess
 import sys
 
 import colorama
+from six.moves import input
 from tockloader import tab
 from tockloader import tbfh
 from tockloader import tockloader as loader
 from tockloader.exceptions import TockLoaderException
 
-# This structure allows us in the future to also support out-of-tree boards.
+PROGRAMMERS = frozenset(("jlink", "openocd", "pyocd", "nordicdfu", "none"))
+
+# This structure allows us to support out-of-tree boards as well as (in the
+# future) more achitectures.
+OpenSKBoard = collections.namedtuple(
+    "OpenSKBoard",
+    [
+        # Location of the Tock board (where the Makefile file is
+        "path",
+        # Targeted architecture (e.g. thumbv7em-none-eabi)
+        "arch",
+        # Size of 1 page of flash memory
+        "page_size",
+        # Flash address at which the kernel will be written
+        "kernel_address",
+        # Set to None is padding is not required for the board
```

Comment:
Might be worth being more specific in what padding we're talking about. Padding starts in flash from this address, which must be the same as `prog` in the board's `layout.ld`, and until `app_address`.

Reply:
Done